ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/helpers/Predicates.java

```diff
@@ -42,29 +42,31 @@ public Predicates(final SpecConfig specConfig) {
    * @see <a
    *     href="https://github.com/ethereum/consensus-specs/blob/v0.8.0/specs/core/0_beacon-chain.md#is_active_validator">is_active_validator</a>
    */
-  public boolean isActiveValidator(Validator validator, UInt64 epoch) {
+  public boolean isActiveValidator(final Validator validator, final UInt64 epoch) {
     return isActiveValidator(validator.getActivationEpoch(), validator.getExitEpoch(), epoch);
   }
 
-  public boolean isActiveValidator(UInt64 activationEpoch, UInt64 exitEpoch, UInt64 epoch) {
+  public boolean isActiveValidator(
+      final UInt64 activationEpoch, final UInt64 exitEpoch, final UInt64 epoch) {
     return activationEpoch.compareTo(epoch) <= 0 && epoch.compareTo(exitEpoch) < 0;
   }
 
   public boolean isValidMerkleBranch(
       final Bytes32 leaf,
       final SszBytes32Vector branch,
       final int depth,
-      int index,
+      final int index,
```

Comment:
Is this actually a legit case in which we don't want it to be `final`?
since this is a primitive type I don't see why we shouldn't allow it.

Reply:
if the error-prone plugin will be designed to enforce it, maybe we can have a custom annotation signalling we actually want it to be mutable (may only for primitive types)

Reply:
yeh does look like we need index to mutate through this fn... pretty unusual!

Reply:
looking at where its used though, i dont htink we actually need to persist the change up, so this one might be fine...

Reply:
So for this case (and I think I found another somewhere else so far 2 out of thousands) there is mutation need within the method and although we could let these be exceptions people who aren't familiar with the code and its workflow might have no context of why and how this is safe to be used due to it's mutation capabilities but so a defensive copy to mutate is probably a safer approach, IMO.

Reply:
Worse case scenario there's always the good old `@SuppressWarnings` but I'd like to avoid using those as much as we can.

Reply:
thinking about it again, I see the safety\readability argument to make a local copy of it, which signal that the value we end up having there will be different. For my preference i'd call it `localIndex` or `indexTmp`, or call the param `startingIndex` and leave the local as `index`.

Reply:
Variables renamed